portal-impl/src/com/liferay/portal/service/impl/WebsiteLocalServiceImpl.java

```diff
@@ -30,10 +31,15 @@ public class WebsiteLocalServiceImpl extends WebsiteLocalServiceBaseImpl {
 
 	@Override
 	public Website addWebsite(
-			long userId, String className, long classPK, String url,
-			long listTypeId, boolean primary, ServiceContext serviceContext)
+			String externalReferenceCode, long userId, String className,
+			long classPK, String url, long listTypeId, boolean primary,
+			ServiceContext serviceContext)
 		throws PortalException {
 
+		if (Validator.isBlank(externalReferenceCode)) {
+			externalReferenceCode = null;
+		}
+
```

Comment:
please remove this (I removed all of these in a previous PR because not needed): the `websitePersistence.update(` already manages the empty space